libs/prelude/Prelude/Interpolation.idr

```diff
@@ -0,0 +1,27 @@
+module Prelude.Interpolation
+
+import Prelude.Ops
+import Prelude.Types
+
+public export
+interface Interpolation a where
+  interpolate : a -> String
+
+||| Concatenation of two things with an interpolation strategy.
+||| Interpolated strings are of the form `"xxxx \{ expr } yyyy"`.
+||| In this example the string `"xxxx "` is concatenated with `expr` and
+||| `" yyyy"`, since `expr` is not necessarily a string, the generated
+||| code looks like this:
+||| ```
+||| interpolate "xxxx " :++: interpolate expr :++: interpolate " yyyy"
```

Comment:
nit: fell out of date with change away from `:++:` operator.

Reply:
ah yep will fix

Reply:
I noticed the old documentation was wrong anyways, it said `interpolate "xxx" :++: interpolate expr` instead of `interpolate "xxx" ++ interpolate expr` or `"xxx " :++: expr`